apps/election-manager/src/config/types.ts

```diff
@@ -79,27 +78,35 @@ export interface ContestTally {
   tallies: ContestOptionTally[]
 }
 
-// TODO: separate into PrecinctTally and ScannerTally
+export interface ContestTallyMeta {
+  overvotes: number
+  undervotes: number
+  ballots: number
+}
+export type ContestTallyMetaDictionary = Dictionary<ContestTallyMeta>
+
 export interface Tally {
   precinctId?: string
   scannerId?: string
+  numberOfBallotsCounted: number
+  castVoteRecords: Map<string, CastVoteRecord>
   contestTallies: ContestTally[]
+  contestTallyMetadata: ContestTallyMetaDictionary
+}
+
+export enum TallyCategory {
+  Precinct = 'precinct',
+  Scanner = 'scanner',
 }
 
 export interface FullElectionTally {
-  scannerTallies: Dictionary<Tally>
-  precinctTallies: Dictionary<Tally>
   overallTally: Tally
+  resultsByCategory: Map<TallyCategory, Dictionary<Tally>>
 }
```

Comment:
I think all these tally-related types should be as read-only as possible. I have a PR that illustrates this for your consideration: https://github.com/votingworks/vxsuite/pull/211